### DISCUSSION TITLE
Feature request: Command to close all workspaces except for the current

### DISCUSSION THREAD

**Discussion by mameen-omar** [nikitabobko/AeroSpace#1021] [feature-ideas]
### Ask
To implement a command to close all workspaces except for the current. 
When invoked, all workspaces except for the focused workspace should be closed and all windows on other workspaces should be moved to the current workspace.

**Comment by nikitabobko** [comment]
**Q:** What is your use case? What workflow do you find this command useful for?

**(1)** I agree that it should be possible to script this behavior. Right now, it's not possible to script it at all because of #173 . And even if #173 were implemented it would not be pleasant to script, because `aerospace` lacks some useful flags #186

Once #186 is implemented it will be possible to script the desired workflow this way:
```
focused=$(aerospace list-workspaces --focused)
aerospace list-windows --all --format window-id | xargs -n1 aerospace move-node-to-workspace $focused --follow-focus --window-id
```

**(2)** I respect the "batteries included approach". That's why I added `close-all-windows-but-current` and `flatten-workspace-tree` commands (btw the former should be renamed to smth like `close-other-windows`)

But I'm not convinced that the command to move all windows to the current workspace would be popular enough to include this battery.

How one would call this command? `move-other-windows-to-workspace <workspace>`? But it could be perceived as "moving other windows of the current workspace". A flag like `--scope (workspace|all-workspaces)` could be introduced. This way `close-other-windows` could be made possible to close windows on other workspaces. This actually sounds interesting, but I need more time to think about it. Other thoughts are welcome

**Comment by XA21X** [comment]
> Q: What is your use case? What workflow do you find this command useful for?

My use case: I often don't want to use workspaces when I don't have any external monitors (normally have three). When I use the dock (or cmd-tab) to switch between apps, the workspace changes automatically, but I can't easily cycle between apps using the focus bindings, I have to remember which workspace I previously had the app/window. Thanks.
